learn-template.yml

```diff
@@ -0,0 +1,1618 @@
+                key: db_username
+          - name: "CONFIG_whisk_couchdb_password"
+            valueFrom:
+              secretKeyRef:
+                name: db.auth
+                key: db_password
+          - name:  "CONFIG_whisk_couchdb_protocol"
+            valueFrom:
+              configMapKeyRef:
+                name: db.config
+                key: db_protocol
+          - name: "CONFIG_whisk_couchdb_host"
+            value: "$(COUCHDB_SERVICE_HOST)"
+          - name: "CONFIG_whisk_couchdb_port"
+            value: "$(COUCHDB_SERVICE_PORT_COUCHDB)"
+          - name: "CONFIG_whisk_couchdb_provider"
+            valueFrom:
+              configMapKeyRef:
+                name: db.config
+                key: db_provider
+          - name: "CONFIG_whisk_couchdb_databases_WhiskActivation"
+            valueFrom:
+              configMapKeyRef:
+                name: db.config
+                key: db_whisk_activations
+          - name: "CONFIG_whisk_couchdb_databases_WhiskEntity"
+            valueFrom:
+              configMapKeyRef:
+                name: db.config
+                key: db_whisk_actions
+          - name: "CONFIG_whisk_couchdb_databases_WhiskAuth"
+            valueFrom:
+              configMapKeyRef:
+                name: db.config
+                key: db_whisk_auths
+  
+          # must match port used in livenessProbe below
+          - name: "PORT"
+            value: "8080"
+  
+          livenessProbe:
+            httpGet:
+              path: "/ping"
+              port: 8080
+              scheme: "HTTP"
+            initialDelaySeconds: 10
+            periodSeconds: 10
+            timeoutSeconds: 10
+            failureThreshold: 5
+          readinessProbe:
+            httpGet:
+              path: "/ping"
+              port: 8080
+              scheme: "HTTP"
+            initialDelaySeconds: 10
+            periodSeconds: 5
+            timeoutSeconds: 10
+            failureThreshold: 5
+        initContainers:
+        - name: wait-for-services
+          image: busybox:${PROJECTODD_LEARN_TAG}
+          command: ['sh', '-cu', 'until nslookup couchdb && nslookup strimzi-openwhisk-kafka; do echo waiting for services; sleep 1; done']
+        - name: wait-for-couchdb
+          image: busybox:${PROJECTODD_LEARN_TAG}
+          command: ['sh', '-c', 'until wget -T 5 --spider http://${COUCHDB_SERVICE_HOST}:${COUCHDB_SERVICE_PORT}/${DB_WHISK_ACTIVATIONS}; do echo waiting for couchdb; sleep 2; done;']
+          env:
+          - name: "DB_WHISK_ACTIVATIONS"
+            valueFrom:
+              configMapKeyRef:
+                name: db.config
+                key: db_whisk_activations
+- apiVersion: batch/v1
+  kind: Job
+  metadata:
+    name: install-catalog
+  spec:
+    activeDeadlineSeconds: 600
+    template:
+      metadata:
+        name: install-catalog
+      spec:
+        containers:
+        - name: catalog
+          image: catalog:${PROJECTODD_LEARN_TAG}
```

Comment:
Jim Crossley, can you tell me what this `catalog` image is for.  We already have a `whisk_catalog` image, but I can seem to find this `catalog` image on any docker hub repo.  So, I'm not quite sure what I should specify for the image name.  Was is supposed to be `whisk_catalog`?

Reply:
ping @jcrossley3 since i'm not sure a specific alert went out with this comment.  Probably @bbrowning , too.

Reply:
That is a reference to the `whisk_catalog` ImageStream. If you're removing ImageStreams, you'd convert this back to `whisk_catalog`. You can look at the history of `template.yml` to see what needs to change if you convert from ImageStreams back to regular images.